sdjournal/read.go

```diff
@@ -161,7 +161,9 @@ process:
 			return ErrExpired
 		default:
 			if c > 0 {
-				writer.Write(msg[:c])
+				if _, err := writer.Write(msg[:c]); err != nil {
```

Comment:
With `:=`, the inner "err" variable masks the outer "err" variable and the function will not return any error. Is it intended and/or correct?